packages/core/src/Page/PageVueServerRenderer.js

```diff
@@ -0,0 +1,132 @@
+ * render the page during Vue mounting.
+ *
+ * This is to avoid the overhead of compiling the page into Vue application
+ * on the client's browser (alleviates FOUC). It is also the pre-requisite to enable SSR.
+ *
+ * @param content Page content to be compiled into Vue app
+ */
+async function compileVuePageAndCreateScript(content, pageConfig, pageAsset) {
+  // Compile Vue Page
+  const compiled = VueCompiler.compileToFunctions(`<div id="app">${content}</div>`);
+
+  // Set up script content
+  const outputContent = `
+    var pageVueRenderFn = ${compiled.render};
+    var pageVueStaticRenderFns = [${compiled.staticRenderFns}];
+  `;
+
+  // Get script file name
+  const pageHtmlFileName = path.basename(pageConfig.resultPath, '.html');
+  const scriptFileName = `${pageHtmlFileName}.page-vue-render.js`;
+
+  /*
+   * Add the script file path for this page's render function to the page's assets (to populate page.njk).
+   * The script file path is the same as the page's file path.
+   */
+  pageAsset.pageVueRenderJs = scriptFileName;
+
+  // Get script's absolute file path to output script file
+  const dirName = path.dirname(pageConfig.resultPath);
+  const filePath = path.join(dirName, scriptFileName);
+
+  await fs.outputFile(filePath, outputContent);
+}
+
+/**
+ * Renders Vue page app into html string (Vue SSR).
+ * This function will install the MarkBindVue plugin and render the built Vue page content into html string.
+ */
+async function renderVuePage(content) {
+  /*
+   * Each installation of plugin pollutes the global scope of Vue.
+   * Thus, we use a fresh copy of Vue each time we install MarkBindVue to prevent the global pollution.
+   * This prevents the old plugin from affecting the new plugin installation (in development mode).
+   */
+  const FreshVue = Vue.extend();
+  FreshVue.use(bundle.MarkBindVue);
+
+  const VueAppPage = new FreshVue({
+    template: `<div id="app">${content}</div>`,
+  });
+
+  const renderedContent = await renderToString(VueAppPage);
+
+  return renderedContent;
+}
+
+/**
+ * Referenced from stackOverflow:
+ * https://stackoverflow.com/questions/17581830/load-node-js-module-from-string-in-memory
+ *
+ * Credits to Dominic
+ */
+function requireFromString(src, filename) {
+  const m = new module.constructor();
+  m.paths = module.paths; // without this, we won't be able to require Vue in the string module
+  m._compile(src, filename);
+  return m.exports;
+}
+
+/**
+ * Retrieves the latest updated MarkBindVue bundle from webpack compiler watcher,
+ * re-render all the built pages, and output the page html files.
+ * This function will only be used in development mode (for MarkBindVue bundle hot-reloading purposes).
+ */
+async function updateMarkBindVueBundle(newBundle) {
+  logger.info(`CHANGES DETECTED IN MarkBindVue COMPONENTS [Update Count: ${updateCount}]
+Bundle is regenerated by webpack and built pages are re-rendered with the latest bundle.`);
+
+  updateCount += 1;
```

Comment:
😅  I used this to help me verify that the development workflow was working properly (that the bundle was getting updated correctly). I left it in as I thought that it may be somewhat useful but honestly it's not necessary. Should I remove it?

Reply:
👍 Let's do that. Not critical as its a development side message.